### PR TITLE
Fix local test compatibility with go 1.18

### PR DESCRIPTION
**Pull request by raptorsun**
This PR fix a compatibility problem in the local test.
The Prometheus used for local test is of version 2.4, 2.19 and 2.23. These versions does not work correctly when compiled with golang 1.18. For example, in 2.4 the API endpoint /api/v1/metadata is not returning anything. So the Prometheus for local test is upgraded to the latest version 2.36.2.

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

The Prometheus instance for the local test is upgraded to the latest version 2.36.2, alongside unit test adjustments.
Old versions are removed from the local tests: 2.4, 2.19, 2.23.

## Verification

Check `make test-local`.

### CODE REVIEW

**Review by bwplotka — DISMISSED** [review]
Amazing job, thanks!

Do you mind fixing DCO?

**Comment by raptorsun** [comment]
> Amazing job, thanks!
> 
> Do you mind fixing DCO?

The commit has been signed to fix DCO errors :)

File: .bingo/Variables.mk [hunk 1]
```diff
@@ -113,15 +113,11 @@ $(PROMDOC): $(BINGO_DIR)/promdoc.mod
...
-	@cd $(BINGO_DIR) && $(GO) build -mod=mod -modfile=prometheus.mod -o=$(GOBIN)/prometheus-v2.4.3+incompatible "github.com/prometheus/prometheus/cmd/prometheus"
-	@echo "(re)installing $(GOBIN)/prometheus-v1.8.2-0.20200724121523-657ba532e42f"
-	@cd $(BINGO_DIR) && $(GO) build -mod=mod -modfile=prometheus.1.mod -o=$(GOBIN)/prometheus-v1.8.2-0.20200724121523-657ba532e42f "github.com/prometheus/prometheus/cmd/prometheus"
-	@echo "(re)installing $(GOBIN)/prometheus-v1.8.2-0.20210331101223-3cafc58827d1"
-	@cd $(BINGO_DIR) && $(GO) build -mod=mod -modfile=prometheus.2.mod -o=$(GOBIN)/prometheus-v1.8.2-0.20210331101223-3cafc58827d1 "github.com/prometheus/prometheus/cmd/prometheus"
+PROMETHEUS_ARRAY := $(GOBIN)/prometheus-v2.36.2
+$(PROMETHEUS_ARRAY): $(BINGO_DIR)/prometheus.mod
+	@# Install binary/ries using Go 1.14+ build command. This is using bwplotka/bingo-controlled, separate go module with pinned dependencies.
+	@echo "(re)installing $(GOBIN)/prometheus-v2.36.2"
+	@cd $(BINGO_DIR) && $(GO) build -mod=mod -modfile=prometheus.mod -o=$(GOBIN)/prometheus-v2.36.2 "github.com/prometheus/prometheus/cmd/prometheus"
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply .bingo/Variables.mk:120]: I just don't understand this comment, otherwise LGTM

[the remark this answers is not in the document]
Comment (raptorsun) [reply .bingo/Variables.mk:120]: $GO is deduced from the env var $PATH by `GO ?= $(shell which go)` at line 6.
Locally I tested with go 1.18 and 1.17, it works for both versions. Earlier versions are not tested. 
The comment, in fact, I have not modified at all :) 
Shall I change the comment to `Install binary/ries using Go 1.17+ build command` ?

**Pull request closed without merging**